src/main/java/com/worksap/nlp/sudachi/PathRewritePlugin.java

```diff
@@ -109,12 +109,13 @@ public LatticeNode concatenate(List<LatticeNodeImpl> path, int begin, int end, L
         StringBuilder readingForm = new StringBuilder();
         for (int i = begin; i < end; i++) {
             LatticeNodeImpl node = path.get(i);
-            surface.append(node.getSurface());
+            StringsCache sc = node.getStrings();
```

Comment:
Since StringsCache is an internal implementation of how LatticeNode holds information, it would be better to ask LatticeNode directly from the outside.